frame/dutch-auction/src/lib.rs

```diff
@@ -110,204 +57,64 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {}
 
-	/// auction can span several dex orders within its lifetime
 	#[derive(Encode, Decode, Default, TypeInfo)]
-	pub struct Order<DexOrderId, AccountId, AssetId, Balance, GroupId> {
-		/// when auction was created(started)
-		pub started: Timestamp,
-		/// how price decreases with time
-		pub function: AuctionStepFunction,
-		/// account who asked for auction and who owns amount to be sold
-		pub account_id: AccountId,
-		/// asset type desired to be sold
-		pub source_asset_id: AssetId,
-		/// account which holds amount to sell
-		pub source_account: AccountId,
-		/// asset type auction wants to get in exchange eventually
-		pub target_asset_id: AssetId,
-		/// account of desired(wanted) currency type to transfer amount after exchange
-		pub target_account: AccountId,
-		/// amount of source currency
-		pub source_total_amount: Balance,
-		/// price of source unit to start auction with.
-		pub source_initial_price: PriceStructure<GroupId, Balance>,
-		/// auction state
-		pub state: AuctionState<DexOrderId>,
+	pub struct TakeBy<AccountId, Balance> {
+		pub from_to: AccountId,
+		pub take: Take<Balance>,
 	}
 
-	type OrderIdOf<T> = <<T as Config>::Orderbook as Orderbook>::OrderId;
-
-	type OrderOf<T> = Order<
-		OrderIdOf<T>,
-		<T as frame_system::Config>::AccountId,
-		<T as DeFiComposableConfig>::AssetId,
-		<T as DeFiComposableConfig>::Balance,
-		<T as Config>::GroupId,
-	>;
-
-	#[pallet::storage]
-	#[pallet::getter(fn orders)]
-	pub type Orders<T: Config> = StorageMap<_, Twox64Concat, T::OrderId, OrderOf<T>, ValueQuery>;
-
 	#[pallet::storage]
 	#[pallet::getter(fn orders_index)]
 	pub type OrdersIndex<T: Config> = StorageValue<_, T::OrderId, ValueQuery>;
 
-	impl<T: Config + DeFiComposableConfig> DutchAuction for Pallet<T> {
-		type AccountId = T::AccountId;
+	#[derive(Encode, Decode, Default, TypeInfo)]
+	pub struct SellOrder<OrderId, AssetId, Balance, AccountId> {
+		pub id: OrderId,
+		pub order: Sell<AssetId, Balance>,
+		// allow to sell by parts like Order Book does, sorted by best proposition
+		pub takes: Vec<TakeBy<AccountId, Balance>>,
```

Comment:
Can we use [BoundedVec](https://crates.parity.io/frame_support/storage/bounded_vec/struct.BoundedVec.html) instead of `Vec` now? It ensure the limit when decoding and when pushing etc...

Reply:
yes. will do.